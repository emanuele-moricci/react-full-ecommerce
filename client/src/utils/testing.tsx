import { FC } from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { BrowserRouter } from "react-router-dom";

export const fakePayloadAction = { type: "FAKE_PAYLOAD" };

export const getMockedStoreRenderedComponent = (state: any, Component: FC) => {
  const mockStore = configureStore();
  const store = mockStore(state);

  return renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Component />
      </BrowserRouter>
    </Provider>
  );
};
