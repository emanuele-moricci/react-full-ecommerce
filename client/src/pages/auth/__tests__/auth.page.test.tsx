import { getMockedStoreRenderedComponent } from "src/utils/testing";

import Auth from "src/pages/auth/auth.page";

describe("<Auth />", () => {
  it("Should match the snapshot", () => {
    const component = getMockedStoreRenderedComponent({}, Auth);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
