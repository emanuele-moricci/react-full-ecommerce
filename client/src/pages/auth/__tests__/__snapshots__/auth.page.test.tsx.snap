// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> Should match the snapshot 1`] = `
<div
  className="sc-kDTinF erjZyr"
>
  <div
    className="sc-eCImPb jnotZz"
  >
    <h2
      className="sc-jRQBWg hRSpAi"
    >
      I already have an account
    </h2>
    <span>
      Sign in with your email and password
    </span>
    <form
      className="sc-iCfMLu"
      onSubmit={[Function]}
    >
      <div
        className="sc-bdvvtL hKCIAl"
      >
        <input
          className="sc-gsDKAQ jbRPqg"
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
        <label
          className="sc-dkPtRN hdMbdV"
        >
          Email
        </label>
      </div>
      <div
        className="sc-bdvvtL hKCIAl"
      >
        <input
          className="sc-gsDKAQ jbRPqg"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
        <label
          className="sc-dkPtRN hdMbdV"
        >
          Password
        </label>
      </div>
      <div
        className="sc-gKclnd jJEzqN"
      >
        <button
          className="sc-hKwDye loXgGx"
          type="submit"
        >
          Sign In
        </button>
        <button
          className="sc-hKwDye hHVCIZ"
          onClick={[Function]}
          type="button"
        >
          Sign In with Google
        </button>
      </div>
    </form>
  </div>
  <div
    className="sc-furwcr jlOgjx"
  >
    <h2
      className="sc-pVTFL ikVURh"
    >
      I do not have an account
    </h2>
    <span>
      Sign up with your email and password
    </span>
    <form
      className="sc-jrQzAO"
      onSubmit={[Function]}
    >
      <div
        className="sc-bdvvtL hKCIAl"
      >
        <input
          className="sc-gsDKAQ jbRPqg"
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          className="sc-dkPtRN hdMbdV"
        >
          Name
        </label>
      </div>
      <div
        className="sc-bdvvtL hKCIAl"
      >
        <input
          className="sc-gsDKAQ jbRPqg"
          name="email"
          onChange={[Function]}
          type="email"
          value=""
        />
        <label
          className="sc-dkPtRN hdMbdV"
        >
          Email
        </label>
      </div>
      <div
        className="sc-bdvvtL hKCIAl"
      >
        <input
          className="sc-gsDKAQ jbRPqg"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <label
          className="sc-dkPtRN hdMbdV"
        >
          Password
        </label>
      </div>
      <div
        className="sc-bdvvtL hKCIAl"
      >
        <input
          className="sc-gsDKAQ jbRPqg"
          name="confPwd"
          onChange={[Function]}
          type="password"
          value=""
        />
        <label
          className="sc-dkPtRN hdMbdV"
        >
          Confirm Password
        </label>
      </div>
      <button
        className="sc-hKwDye loXgGx"
        type="submit"
      >
        SIGN UP
      </button>
    </form>
  </div>
</div>
`;
